frontend/catalyst/device/qjit_device.py

```diff
@@ -466,13 +466,17 @@ def preprocess(
         backend according to the backend TOML file).
         """
 
+        if execution_config is None:
+            execution_config = qml.devices.ExecutionConfig()
+
         _, config = self.original_device.preprocess(execution_config)
 
         program = TransformProgram()
 
         # measurement transforms may change operations on the tape to accommodate
         # measurement transformations, so must occur before decomposition
         measurement_transforms = self._measurement_transform_program()
+        config = replace(config, device_options=deepcopy(config.device_options))
```

Comment:
In most cases (when no `execution_config` argument is given), this will lead to an extra copy operation, but it protects against modifying the global `DefaultExecutionConfig` in case it ever gets passed in here.